### PR TITLE
Refactor/test version config

### PR DESCRIPTION
**Pull request by Mpdreamz**
- Simplifies `ITestConfiguration` and its implementations. 
- Default version resolutions is now driven by the yaml file rather than also having baked in fallback in code.

### CODE REVIEW

File: build/scripts/Testing.fs [hunk 1]
```diff
@@ -54,7 +53,10 @@ module Tests =
             | (true) -> [ "--logger"; "trx"; "--collect"; "\"Code Coverage\""; "-v"; "m"] |> List.append command
             | _  -> command
             
-        Tooling.DotNet.ExecInWithTimeout "src/Tests/Tests" commandWithCodeCoverage (TimeSpan.FromMinutes 30.) 
+        // using std redirection since that foces vstest not to redirect Console.Write from Elastic.Xunit somehow
```
Comment (russcam) [inline build/scripts/Testing.fs]: Should this comment be updated with how vstest and xunit communicate?

File: src/Tests/Tests.Configuration/TestConfigurationBase.cs [hunk 2]
```diff
@@ -1,22 +1,70 @@
...
+		/// <summary> whether the seed was provided externally to be fixated </summary>
+		public bool SeedProvidedExternally { get; private set; }
+
+		/// <summary>
+		/// This is fixed for now, specifying false leads to flaky tests, warrants deeper investigation
+		/// in our abstractions project
+		/// </summary>
+		public bool ShowElasticsearchOutputAfterStarted { get; } = true;
+
+		/// <summary> When specified will only run one overload in API tests, helpful when debugging locally </summary>
```
Comment (russcam) [inline src/Tests/Tests.Configuration/TestConfigurationBase.cs]: comments 💯

File: build/scripts/Testing.fs (hunk 1 above)
Comment (Mpdreamz) [reply build/scripts/Testing.fs]: reverting this since `Proc` is not happy on linux

**Comment by Mpdreamz** [comment]
ported to master

**Pull request merged**